src/Engine/ProtoCore/ClassTable.cs

```diff
@@ -393,7 +393,7 @@ public class ClassTable
         //Symbol table to manage symbols with namespace
         private Namespace.SymbolTable symbolTable = new Namespace.SymbolTable();
         
-        private List<ClassNode> GetClassHierarchy(ClassNode node)
+        public List<ClassNode> GetClassHierarchy(ClassNode node)
```

Comment:
Does `internal` work here?

Reply:
No, since this method is defined in `ProtoCore` but now being used in `DynamoCore`.

Reply:
I understand this is being used by dynamo core - we also can use the internalsVisibleTo attribute to expose internal members to other assemblies which we own.

We should develop this principal further, but a few points.
* ProtoCore.dll is currently shipped as a part of the DynamoCore nuget package. To me looking at this package, as an external developer I see this as part of the API.
* A public method in this assembly which is shipped as part of a nuget package, will end up needing to be maintained.

Is this method something that should be used externally by integrators? Is there a compelling reason to make it part of the API? An interesting use case? A workflow that cannot be implemented without it? 

If not, then what is the benefit?

If yes, then great, let's add documentation to it.

Reply:
Have we made a decision on going forward with reducing the API surface of DynamoCore, DynamoCoreWpf etc. by continuing to expose them as our public API or are we deciding on keeping them as-is and making them internal and having a separate API layer for public consumption (based on your RFC)? 

Whatever is the strategy, I think then we should come up with certain coding guidelines before we continue to make PR's. Going by the RFC again, it sounds like none of ProtoCore should be part of the API and we should surface just the AST's publicly. I'll make this method internal as well for now.

Reply:
No, we have made no decision -very much welcome those discussion - there are many options.
Thanks for addressing this, I don't think we can stop all work before we entirely decide this - but what  I am suggesting is that we proceed conservatively until we do so. 

I think it's likely we will find that some users are *already* using ProtoCore, and other binaries we didn't consider fully as APIs.

For example, the Civil3d integration has very specific interactions with callsites and trace, I am not 100% what APIs that team uses to control and redirect trace but I guess it is in protocore.